src/Illuminate/Mail/MailManager.php

```diff
@@ -135,32 +133,15 @@ protected function resolve($name)
         return $mailer;
     }
 
-    /**
-     * Create the SwiftMailer instance for the given configuration.
-     *
-     * @param  array  $config
-     * @return \Swift_Mailer
-     */
-    protected function createSwiftMailer(array $config)
-    {
-        if ($config['domain'] ?? false) {
-            Swift_DependencyContainer::getInstance()
-                ->register('mime.idgenerator.idright')
-                ->asValue($config['domain']);
```

Comment:
Swift-Mailer generated the Message-ID with `xxx@swift.generated` you could change the `swift.generated` part with the domain like you did.

In Mime-Component of Symfony the Message-ID is generated from the `sender` address.
Example: If the sender address is `info@laravel.com` then the generated ID would be `xxx@laravel.com`
https://github.com/symfony/mime/blob/8b6fc5f7323a046dfee458169c62a3ec42294f56/Message.php#L138

This Message-ID is then added to SentMessage.
https://github.com/symfony/mailer/blob/088e5b21d82d7cf7e3e6bf0d8636d3b45b9e33f9/SentMessage.php#L31

There would probably no need for further configuration. Because it is already customised.

Reply:
Thanks for figuring that out @Jubeki 👍